draft-dt-tls-external-psk-guidance.md

```diff
@@ -333,7 +334,21 @@ PSK identities as usernames.
 - Deployments should take care that the length of the PSK identity is sufficient
 to avoid obvious collisions.
 
-[[OPEN ISSUE: discuss implication of collisions between  external and resumption PSKs.]]
+### PSK Identity Collisions
+
+It is possible, though unlikely, that an external PSK identity may clash with a
+resumption PSK identity. The TLS stack implementation and sequencing of PSK callbacks
+influences the application's behaviour when identity collisions occur. When a server
+receives a PSK identity in a TLS 1.3 ClientHello, OpenSSL, BoringSSL and mbedTLS 
+execute the application's registered callback function before checking the stack's
+internal session resumption cache. This means that if a PSK identity collision occurs,
+the application will be given precedence over how to handle the PSK.
+
+[[TODO.. the application still does not know if an identity collision occured or not.
+If we wanted to be 100% explicit, the stack would check its resumption cache and then
+callback the application indicating here is a PSK that has a cache hit, please check
+your external PSK DB, and let me know how to proceed. Seems like overkill for something
+that should not happen.]]
```

Comment:
Yes, this seems like overkill since it's an implementation detail. I'll remove this TODO after merging.